Sources/OTelTesting/OTelInMemorySpanProcessor.swift

```diff
@@ -0,0 +1,43 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift OTel open source project
+//
+// Copyright (c) 2023 Moritz Lang and the Swift OTel project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import OpenTelemetry
+import ServiceContextModule
+
+/// An in-memory span processor, collecting started spans into ``OTelInMemorySpanProcessor/startedSpans``
+/// and finished spans into ``OTelInMemorySpanProcessor/finishedSpans``.
+public final actor OTelInMemorySpanProcessor: OTelSpanProcessor {
+    public private(set) var startedSpans = [(span: OTelSpan, parentContext: ServiceContext)]()
+    public private(set) var finishedSpans = [OTelFinishedSpan]()
+    public private(set) var numberOfForceFlushes = 0
+    public private(set) var numberOfShutdowns = 0
+
+    public init() {}
+
+    public func run() async throws {
+        while !Task.isCancelled {}
```

Comment:
At least maybe put a sleep here? This'll intensely burn CPU

Reply:
Yeah a long sleep is fine -- don't do MAX_VALUE seconds since we have a bug with that and I'll wake up immediately.

Just a "few hours sleep" will do the job fine here